File: LastLogged/Models/TrackerItem.swift
import Foundation
import SwiftData

@Model
final class TrackerItem {
    @Attribute(.unique) var id: UUID
    var name: String
    var categoryId: UUID
    var reminderIntervalDays: Int?
    var lastCompletedAt: Date?
    var createdAt: Date
    var sortOrder: Int
    var iconName: String
    var isArchived: Bool
    var syncStatus: SyncStatus
    var updatedAt: Date

    init(
        id: UUID = UUID(),
        name: String,
        categoryId: UUID,
        reminderIntervalDays: Int? = nil,
        lastCompletedAt: Date? = nil,
        createdAt: Date = Date(),
        sortOrder: Int = 0,
        iconName: String = "checkmark.circle",
        isArchived: Bool = false,
        syncStatus: SyncStatus = .pending,
        updatedAt: Date = Date()
    ) {
        self.id = id
        self.name = name
        self.categoryId = categoryId
        self.reminderIntervalDays = reminderIntervalDays
        self.lastCompletedAt = lastCompletedAt
        self.createdAt = createdAt
        self.sortOrder = sortOrder
        self.iconName = iconName
        self.isArchived = isArchived
        self.syncStatus = syncStatus
        self.updatedAt = updatedAt
    }
}

File: android/app/src/main/java/com/pearsonmedia/lastlogged/data/local/entity/TrackerItem.kt
package com.pearsonmedia.lastlogged.data.local.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "tracker_items")
data class TrackerItem(
    @PrimaryKey
    val id: String,

    val name: String,

    @ColumnInfo(name = "category_id")
    val categoryId: String? = null,

    @ColumnInfo(name = "reminder_interval_days")
    val reminderIntervalDays: Int = 30,

    @ColumnInfo(name = "last_completed_at")
    val lastCompletedAt: Long? = null,

    @ColumnInfo(name = "created_at")
    val createdAt: Long = System.currentTimeMillis(),

    @ColumnInfo(name = "sort_order")
    val sortOrder: Int = 0,

    @ColumnInfo(name = "icon_name")
    val iconName: String = "checklist",

    @ColumnInfo(name = "is_archived")
    val isArchived: Boolean = false,

    @ColumnInfo(name = "sync_status")
    val syncStatus: SyncStatus = SyncStatus.PENDING,

    @ColumnInfo(name = "updated_at")
    val updatedAt: Long = System.currentTimeMillis()
)

enum class SyncStatus {
    SYNCED,
    PENDING,
    CONFLICT
}
